patina_dxe_core/src/pi_dispatcher/fv.rs

```diff
@@ -0,0 +1,1702 @@
+                    let file_type_read_ref1: *mut fv::EfiFvFileType = &mut file_type_read;
+
+                    MockProtocolData::fv_get_next_file_efiapi(
+                        fv_ptr1,
+                        buffer_valid3,
+                        file_type_read_ref1,
+                        n_guid_ref_mut,
+                        file_attributes,
+                        buffer_valid_size3,
+                    );
+                    /* Null BUffer Case*/
+                    MockProtocolData::fv_get_next_file_efiapi(
+                        fv_ptr1,
+                        std::ptr::null_mut(),
+                        file_type_read_ref,
+                        n_guid_ref_mut,
+                        file_attributes,
+                        buffer_valid_size3,
+                    );
+                    // Deallocate the memory
+                    dealloc(buffer_valid3 as *mut u8, layout3);
+                };
+
+                let fvb_test_read_section = || {
+                    /* Mutable Reference cannot be borrowed more than once,
+                     * hence delcare and free up after use immediately
+                     */
+                    let mut len3 = 1000;
+                    let buffer_valid_size3: *mut usize = &mut len3;
+                    let layout3 = Layout::from_size_align(1001, 8).unwrap();
+                    let mut buffer_valid3 = alloc(layout3) as *mut c_void;
+
+                    if buffer_valid3.is_null() {
+                        panic!("Memory allocation failed!");
+                    }
+
+                    let mut gd2: efi::Guid = efi::Guid::from_fields(
+                        0x434f695c,
+                        0xef26,
+                        0x4a12,
+                        0x9e,
+                        0xba,
+                        &[0xdd, 0xef, 0x00, 0x97, 0x49, 0x7c],
+                    );
+                    let name_guid2: *mut efi::Guid = &mut gd2;
+
+                    /* Cover the NULL Case, User Passing Invalid Parameter Case  */
+                    MockProtocolData::fv_read_section_efiapi(
+                        ptr::null(),
+                        ptr::null(),
+                        SECTION_TYPE,
+                        SECTION_INSTANCE,
+                        std::ptr::null_mut(),
+                        std::ptr::null_mut(),
+                        std::ptr::null_mut(),
+                    );
+
+                    MockProtocolData::fv_read_section_efiapi(
+                        fv_ptr1,
+                        guid_ref_invalid_ref,
+                        6,
+                        10,
+                        &mut buffer_valid3 as *mut *mut c_void,
+                        buffer_valid_size3,
+                        auth_valid_p,
+                    );
+
+                    /* Valid guid case - panicing, debug this further, for now comment*/
```

Comment:
maybe file an issue?

Reply:
Agreed, but if we don't have an issue tracking it, we probably should. I'm happy to file if that would help.

Reply:
I think the comment is stale now?